libpod/runtime_pod_linux.go

```diff
@@ -304,6 +309,12 @@ func (r *Runtime) removePod(ctx context.Context, p *Pod, removeCtrs, force bool,
 		}
 	}
 
+	// let's unlock the containers so if there is any cleanup process, it can terminate its execution
+	for _, ctr := range ctrs {
+		ctr.lock.Unlock()
```

Comment:
Just a small non-blocker concern here: Would this allow other podman process to modify ctrs while the pod is being removed same as https://github.com/containers/podman/pull/13403#discussion_r817596145

Reply:
nvm we are already removing all containers from state before this so I think this should be only useful for another cleanup process.